File: problem-2/efibsum.jl
function efibsum(n, nprev, max)
	if n > max
		return 0
	end

	if n % 2 != 0
		return efibsum(n + nprev, n, max)
	end

	return n + efibsum(n + nprev, n, max)
end

max = parse(Int, ARGS[1])

println(efibsum(1, 1, max))

File: problem-2/efibsum.hs
efibsum :: Int -> Int -> Int -> Int
efibsum n nprev max
    | n > max       = 0
    | mod n 2 /= 0  = efibsum (n + nprev) n max
    | otherwise     = n + efibsum (n + nprev) n max

main = do
    print(efibsum 1 1 4000000)

